service/src/io/pedestal/http/route/definition.clj

```diff
@@ -31,10 +47,20 @@
 (defmethod expand-constraint clojure.lang.APersistentMap [query-constraint-spec]
   (expand-query-constraint query-constraint-spec))
 
+(defmethod expand-constraint clojure.lang.PersistentVector [spec]
+  (assert false (unexpected-vector-in-route spec)))
+
+(defmethod expand-constraint :default [unmatched]
+  (assert false (unmatched-type-in-constraint unmatched)))
+
 (defprotocol ExpandableVerbAction
   (expand-verb-action [expandable-verb-action]
     "Expand `expandable-verb-action` into a verbose-form verb-map."))
 
+(def valid-handler?      (some-fn seq? symbol? interceptor/interceptor?))
+(def interceptor-vector? (every-pred vector? (comp :interceptors meta)))
+(def constraint-map?     (every-pred map?    (comp :contraints   meta)))
```

Comment:
Good catch. I got bitten by recompiling the code that the macro uses, but not the namespace that uses the macro. Grr.